src/main/java/com/nutomic/syncthingandroid/syncthing/EventProcessor.java

```diff
@@ -0,0 +1,171 @@
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import eu.chainfire.libsuperuser.Shell;
+
+/**
+ * Run by the syncthing service to convert syncthing events into local broadcasts.
+ * It uses SyncthingService.GetEvents to read the pending events and wait for new events.
+ */
+public class EventProcessor implements SyncthingService.OnWebGuiAvailableListener, Runnable, RestApi.OnReceiveEventListener {
+
+    private static final String TAG = "EventProcessor";
+    private static final String PREF_LASTSYNCID = "last_sync_id";
+
+    private static final String EVENT_BASE_URL = "com.nutomic.syncthingandroid.event";
+
+    /**
+     * Minimum interval in seconds at which the events are polled from syncthing and processed.
+     * This intervall will not wake up the device to save battery power.
+     */
+    public static final long EVENT_UPDATE_INTERVAL = 15;
+
+    // Use the MainThread for all callbacks and message handling or we have to track down nasty threading problems.
+    private final Handler mMainThreadHandler = new Handler(Looper.getMainLooper());
+
+    private volatile long mLastEventId = 0;
+    private volatile boolean mShutdown = true;
+
+    private final Context mContext;
+    private final RestApi mApi;
+    private final LocalBroadcastManager mLocalBM;
+    private final Map<String, String> mFolderToPath = new HashMap<String, String>();
+
+    public static String getEventUrl(String eventName) {
+        return EVENT_BASE_URL + "." + eventName.toUpperCase();
+    }
+
+    public EventProcessor(Context context, RestApi api) {
+        mContext = context;
+        mApi = api;
+        mLocalBM = LocalBroadcastManager.getInstance(mContext);
+    }
+
+    private void updateFolderMap()
+    {
+        synchronized(mFolderToPath) {
+            mFolderToPath.clear();
+            for (RestApi.Folder folder: mApi.getFolders()) {
+                mFolderToPath.put(folder.id, folder.path);
+            }
+        }
+    }
+
+    @Override
+    public void run() {
+        // Restore the last event id if the event processor may have been restartet.
+        if (mLastEventId == 0) {
+            mLastEventId = PreferenceManager.getDefaultSharedPreferences(mContext).getLong(PREF_LASTSYNCID, 0);
+        }
+
+        // First check if the event number ran backwards. If that's the case we've to start at zero because syncthing was restartet.
+        mApi.getEvents(0, 1, new RestApi.OnReceiveEventListener() {
+            @Override
+            public void onEvent(long id, String eventType, Bundle eventData) {
+            }
+
+            @Override
+            public void onDone(long lastId) {
+                // Now we know the last event ID. We check if it ran backwards and then start loading events.
+                if (lastId < mLastEventId) mLastEventId = 0;
```

Comment:
You basically have the same comment twice, in line 83 and 91 :p 

Also, can you make the code after the if in a new line for clarity/consistency? (same in line 107).